extended_mypy_django_plugin/_plugin/analyze.py

```diff
@@ -138,21 +142,24 @@ def transform_cast_as_concrete(self, ctx: DynamicClassDefContext) -> None:
                     # a variable typed in terms of Self
                     func.type.arg_types[0] = replacement
 
+        # TODO: name to indicate that concrete is a type.
         concrete = self.make_resolver(ctx=ctx).resolve(
             protocols.KnownAnnotations.CONCRETE,
             TypeType(arg_node_typ) if is_type else arg_node_typ,
         )
+        # TODO: In this case the resolver will have already complained.
         if not concrete:
             return None
 
         # Perform a cast!
-        ctx.call.analyzed = CastExpr(ctx.call.args[0], concrete)
+        ctx.call.analyzed = CastExpr(first_arg, concrete)
         ctx.call.analyzed.line = ctx.call.line
         ctx.call.analyzed.column = ctx.call.column
         ctx.call.analyzed.accept(sem_api)
 
         return None
 
+    # TODO: T_Leader = TypeVar("T_Leader", bound=Concrete[Leader])
```

Comment:
I haven't looked into this but I have done the rest of the notes in https://github.com/delfick/extended-mypy-django-plugin/pull/45/commits/78f0a73677cc616e74c831d5a0c5df4a84d1ae15